svcdef/consolidate_http.go

```diff
@@ -0,0 +1,122 @@
+package svcdef
+
+import (
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/TuneLab/go-truss/deftree/svcparse"
+	"github.com/pkg/errors"
+)
+
+// ConsolidateHTTP accepts a SvcDef and the io.Readers for the proto files
+// comprising the definition. It modifies the SvcDef so that HTTPBindings and
+// their associated HTTPParamters are added to each ServiceMethod. After this,
+// each `HTTPBinding` will have a populated list of all the http parameters
+// that that binding requires, where that parameter should be located, and the
+// type of each parameter.
+func ConsolidateHTTP(sd *Svcdef, protoFiles []io.Reader) error {
+	for _, pfile := range protoFiles {
+		lex := svcparse.NewSvcLexer(pfile)
+		protosvc, err := svcparse.ParseService(lex)
+		if err != nil {
+			if strings.Contains("'options' or", err.Error()) {
+				log.Warnf("Parser found rpc method which lacks HTTP " +
+					"annotations; this is allowed, but will result in HTTP " +
+					"transport not being generated.")
+			} else {
+				return errors.Wrap(err, "error while parsing http options for the service definition")
+			}
+		}
+		assembleHTTPParams(sd.Service, protosvc)
+	}
+	return nil
+}
+
+func assembleHTTPParams(svc *Service, httpsvc *svcparse.Service) error {
```

Comment:
`assembleHTTPParams` is named somewhat vaguely, and would probably benefit from a doc-comment explaining its purpose, as well as a few comments providing naration of its actions.